apps/web/drizzle.config.ts

```diff
@@ -1,12 +1,41 @@
-import type { Config } from "drizzle-kit";
-
-export default {
-	schema: "./app/server/db/schema.ts",
-	dialect: "sqlite",
-	driver: "d1",
-	dbCredentials: {
-		wranglerConfigPath: "./wrangler.toml",
-		dbName: "",
-	},
-	out: "migrations",
-} satisfies Config;
+import { defineConfig } from "drizzle-kit";
+import fs from "node:fs";
+import path from "node:path";
+
+function getLocalD1DB() {
+  try {
+    const basePath = path.resolve(".wrangler");
+    const dbFile = fs
+      .readdirSync(basePath, { encoding: "utf-8", recursive: true })
+      .find((f) => f.endsWith(".sqlite"));
+
+    if (!dbFile) {
+      throw new Error(`.sqlite file not found in ${basePath}`);
+    }
+
+    const url = path.resolve(basePath, dbFile);
+    return url;
+  } catch (err) {
+    console.log(`Error  ${err}`);
+  }
+}
+
+export default defineConfig({
+  dialect: "sqlite",
+  schema: "./app/server/db/schema.ts",
+  out: "./drizzle",
+  ...(process.env.NODE_ENV === "production"
+    ? {
+        driver: "d1-http",
+        dbCredentials: {
```

Comment:
these tokens need to either be in the setup script or detailed instructions in the readme or the CLI
But I like this a lot, especially because these are production-only vars! Question: Why not just use the binding in production?

Reply:
There's also a slight latency hit when doing this because each request goes through edge auth